route.go

```diff
@@ -21,6 +21,7 @@ type Route struct {
 	Path     string // webservice root path + described path
 	Function RouteFunction
 	Filters  []FilterFunction
+	If       []RouteIfFunction
```

Comment:
I hesitate whether the name RouteIfFunction is the best fit. Is it clear enough to what process the "If" is related?
Its behaviour is more like a RouteSelectionConditionFunction.

Reply:
I'd be happy to rename. I think that the builder reads better as `.If(...)` but changing the type to `RouteSelectionConditionFunction` does make the purpose and usage clearer.

Reply:
The "If" is ok, it's readable like this.  Please rename the type.